src/fee.rs

```diff
@@ -0,0 +1,25 @@
+/// Function for computing a fee.
+
+/// Get fee in vbytes
+#[cfg(any(test, feature = "rand"))]
+pub fn get_fee_vbytes(fee_rate: u64, weight: u64) -> u64 {
+    //One vbyte is equal to four weight units.
+    let vbytes = weight / 4;
+
+    vbytes * fee_rate
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::fee::*;
+
+    #[test]
+    fn get_effective_value() {
+        let fee_rate = 10;
```

Comment:
If you are going with integers for the feerate, I would recommend using `ṩ/kvB` as the base unit, since it allows you to be more precise than just full satoshi increments. That way you can create transactions with feerates at `1.103 ṩ/vB`, instead of needing to pay `2 ṩ/vB` if `1 ṩ/vB` is insufficient for the priority you want to imbue the transaction with.

Reply:
I was also about to ask about how you avoid dust outputs, but then I saw the last commit. :+1:

Reply:
> If you are going with integers for the feerate, you could consider using ṩ/kvB as the base unit

Thanks, that's a good point.  Wouldn't it be better to use a float for rate?

Otherwise, [here](https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/files#diff-f4eff9ed318d5e5d7acac95241be35055cf36968693da8a60c16b8b53e97fdd4R14) would be: 

`let kv_bytes = weight / 4 /1024;`
which means we lose a lot of precision with int division OR kvbytes is a float.  It doesn't work to have kvbytes be a float and feerate as an int then.

Reply:
We don’t use “kibivirtualbytes”, but “kilovirtualbytes“:

    1 MvB = 1000 kvB = 1 000 000 vB
    
We never use floats for consensus relevant calculations in Bitcoin Core to avoid any sort of rounding issues. YMMV.

Reply:
How about something like this: https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/d1d38b16611e45a57ded93daa715126216156b84

```
/// Takes as input the fee_rate in $ per Kilovirtualbytes
/// and a size in Kilovirtualbytes and returns the fee amount.
///
/// * fee_rate: $ per kilovirtualbytes $KvB
/// * size: kilovirtualbytes KvB
/// * fee: $
```

Reply:
I just noticed this PR https://github.com/rust-bitcoin/rust-bitcoin/pull/1627.  Maybe it would be better to build off `FeeRate` as defined by Rust Bitcoin.

Reply:
SGTM

Reply:
> How about something like this: [d1d38b1](https://github.com/p2pderivatives/rust-bitcoin-coin-selection/commit/d1d38b16611e45a57ded93daa715126216156b84)
> 
> ```
> /// Takes as input the fee_rate in $ per Kilovirtualbytes
> /// and a size in Kilovirtualbytes and returns the fee amount.
> ///
> /// * fee_rate: $ per kilovirtualbytes $KvB
> /// * size: kilovirtualbytes KvB
> /// * fee: $
> ```

It’s not a dollar symbol but an “s” with a dot above and below: ṩ (u+1e69). The symbol for kilo is “k”, not “K”, the latter is the symbol for Kelvin, so it’s kvB, and that makes the feerate ṩ/kvB (which without the slash would be a multiplication rather than a division).